github/github.go

```diff
@@ -137,7 +137,7 @@ var errNonNilContext = errors.New("context must be non-nil")
 // A Client manages communication with the GitHub API.
 type Client struct {
 	clientMu sync.Mutex   // clientMu protects the client during calls that modify the CheckRedirect func.
```

Comment:
Will exposing `client` directly lead to problems with this mutex? I think two different `github.Client` objects would each use separate locks to protect the same `*http.Client` object, leading to race conditions.

Maybe instead there could be a new `Client()` or `HTTPClient()` function that returns a shallow copy of the internal `client` field? That would enable reuse of the transport and timeout while allowing each instance to safely update the redirect function.

Reply:
Hmmm... that's an interesting question...

The idea was to prevent multiple calls that check for redirects to avoid swapping out the client at the same time.

But maybe a read-only `Client()` method as you recommend would be a better solution anyway.